### ISSUE TITLE
CLI Menu: Switch vs Object Literal

### ISSUE THREAD

**Issue by gavbarnett** [ianisted/jquery-cli#6]
Consider looking at this to clean up the CLI menu selection. I think this would make it easier to add new functions and add layers to the menu system. 
I think this also opens up options for _tab-complete_ implementation.

[Switch vs Object Literals](https://toddmotto.com/deprecating-the-switch-statement-for-object-literals/)

Some reading to do but my understanding would be something like:
1. init program
2. pull in BasicFunctions menu
3. wait for user input
4. if user input introduces new menus (say we start playing a game with north, south, east, west options) then add these to menu options.
5. equally menu items can be removed if required.

In summary you have a menu as an array.

**Comment by gavbarnett** [comment]
something like this perhaps:

```
//global CLI_Menu
var CLI_Menu = {'default': BasicFunction.echo("ERROR: Selection not understood")};

function Init(){
    CLI_Menu = {
        //load initial menu
        'echo': BasicFunction.echo(stdin),
        'help': BasicFunction.help(stdin),
        :, //other functions
        :, //
        'default': BasicFunction.echo("ERROR: Selection not understood")  //add a default case
    };
}

function Game(){
    CLI_Menu = {
        //load game menu
       //either as addition (append) or replacement of existing menu
        'north': GameFunction.north(stdin),
        'south': GameFunction.south(stdin),
        :, //other functions
        :, //
        'default': BasicFunction.echo("ERROR: Selection not understood")  //add a default case
    };
}

function Menu_selection(stdin, CLI_Menu) {
    stdout = (CLI_Menu[stdin] || CLI_Menu['default']);  //return with default option
   return stdout;
}

CLI_Menu('echo');   //Call the echo function
```

**Comment by gavbarnett** [comment]
tab-complete could be a bit like (this is pseudo-code-ish just to give the idea):

```
function tab_complete(stdin){
    $.each(CLI_Menu, function(key, value) {
        if (left(key,len(stdin)) = stdin){
         //valid option add to options list
        };
      };
    if (len(option list) = 1){
         stdout = CLI_Menu(optionlist[0])
    }else{
        echo all options;
    };
}
```

**Comment by gavbarnett** [comment]
Could also updated the Help function to use this to show all valid commands at any given time.
- would need to be careful this does not make any "game" aspects too easy. 

i.e 
echo whats the password
input help
echo valid commands are: pa$$W0rD
input pa$$W0rD

**Comment by gavbarnett** [comment]
Changed Switch for Objects in latest push to master!!! 
Really happy with this big change. 
Closing this enhancement. Will open new for tab-complete.
